### PR TITLE
Bugfix/fix salt upgrade downgrade

### PR DESCRIPTION
**Pull request by TeddyAndrieux**
**Component**:

'salt', 'lifecycle'

**Context**: 

Various issue for upgrade and downgrade

**Summary**:

- Special case to handle `salt-minion` downgrade
- Fix `salt-minion` upgrade and downgrade during orchestrate

---

Fixes: #2524

### CODE REVIEW

File: salt/metalk8s/salt/minion/running.sls [hunk 1]
```diff
@@ -4,19 +4,22 @@ Restart salt-minion:
     - bg: true
 
 Wait until salt-minion restarted:
-  module.wait:
-    - test.sleep:
-      - length: 10
-    - watch:
+  test.configurable_test_state:
+    - changes: False
+    - result: __solt__:salt:test.sleep(10)
```
Comment (gdemonet) [inline salt/metalk8s/salt/minion/running.sls]: Typo?

Comment (TeddyAndrieux) [reply salt/metalk8s/salt/minion/running.sls]: Uh ! Yeah,
salt, solt .... you know :)

**Review by gdemonet — DISMISSED** [review]
:shipit:

**Review by gdemonet — APPROVED** [review]

**Pull request merged**